src/main/drivers/bus_i2c_stm32f4xx.c

```diff
@@ -371,115 +433,54 @@ uint16_t i2cGetErrorCounter(void)
-    digitalHi(gpio_sda, sda);
+    GPIO_SetBits(gpio_scl, scl);
+    GPIO_SetBits(gpio_sda, sda);
 
-    cfg.pin = scl;
-    cfg.speed = Speed_2MHz;
-    cfg.mode = Mode_Out_OD;
-    gpioInit(gpio_scl, &cfg);
+    gpio.GPIO_Pin = scl;
+    GPIO_Init(gpio_scl, &gpio);
 
-    cfg.pin = sda;
-    cfg.speed = Speed_2MHz;
-    cfg.mode = Mode_Out_OD;
-    gpioInit(gpio_sda, &cfg);
+    gpio.GPIO_Pin = sda;
+    GPIO_Init(gpio_sda, &gpio);
 
     for (i = 0; i < 8; i++) {
         // Wait for any clock stretching to finish
-        while (!digitalIn(gpio_scl, scl))
+        while (!GPIO_ReadInputDataBit(gpio_scl, scl))
             delayMicroseconds(10);
 
         // Pull low
-        digitalLo(gpio_scl, scl); // Set bus low
+        GPIO_ResetBits(gpio_scl, scl); // Set bus low
         delayMicroseconds(10);
         // Release high again
-        digitalHi(gpio_scl, scl); // Set bus high
+        GPIO_SetBits(gpio_scl, scl); // Set bus high
         delayMicroseconds(10);
     }
 
     // Generate a start then stop condition
-    // SCL  PB10
-    // SDA  PB11
-    digitalLo(gpio_sda, sda); // Set bus data low
+    GPIO_ResetBits(gpio_sda, sda); // Set bus data low
     delayMicroseconds(10);
-    digitalLo(gpio_scl, scl); // Set bus scl low
+    GPIO_ResetBits(gpio_scl, scl); // Set bus scl low
     delayMicroseconds(10);
-    digitalHi(gpio_scl, scl); // Set bus scl high
+    GPIO_SetBits(gpio_scl, scl); // Set bus scl high
     delayMicroseconds(10);
-    digitalHi(gpio_sda, sda); // Set bus sda high
-
-    // Init pins
-    cfg.pin = scl;
-    cfg.speed = Speed_50MHz;
-    cfg.mode = Mode_AF_OD;
-    gpioInit(gpio_scl, &cfg);
-
-    cfg.pin = sda;
-    cfg.speed = Speed_50MHz;
-    cfg.mode = Mode_AF_OD;
-    gpioInit(gpio_sda, &cfg);
-#else
-    GPIO_TypeDef *gpio;
-    gpio_config_t cfg;
-    uint16_t scl, sda;
-    int i;
-
-    // prepare pins
-    gpio = i2cHardwareMap[I2Cx_index].gpio;
-    scl = i2cHardwareMap[I2Cx_index].scl;
-    sda = i2cHardwareMap[I2Cx_index].sda;
-
-    digitalHi(gpio, scl | sda);
-
-    cfg.pin = scl | sda;
-    cfg.speed = Speed_2MHz;
-    cfg.mode = Mode_Out_OD;
-    gpioInit(gpio, &cfg);
-
-    for (i = 0; i < 8; i++) {
-        // Wait for any clock stretching to finish
-        while (!digitalIn(gpio, scl))
-            delayMicroseconds(10);
-
-        // Pull low
-        digitalLo(gpio, scl); // Set bus low
-        delayMicroseconds(10);
-        // Release high again
-        digitalHi(gpio, scl); // Set bus high
-        delayMicroseconds(10);
-    }
+    GPIO_SetBits(gpio_sda, sda); // Set bus sda high
 
-    // Generate a start then stop condition
-    // SCL  PB10
-    // SDA  PB11
-    digitalLo(gpio, sda); // Set bus data low
-    delayMicroseconds(10);
-    digitalLo(gpio, scl); // Set bus scl low
-    delayMicroseconds(10);
-    digitalHi(gpio, scl); // Set bus scl high
-    delayMicroseconds(10);
-    digitalHi(gpio, sda); // Set bus sda high
+    gpio.GPIO_Pin = scl;
```

Comment:
This is not realy needed since the pins will be reinitialized in the i2cInitPort() function to AF mode anyhow. In the previous code this was done here. Since there was no additional pin initialization.